src/core/abr/representation_chooser.ts

```diff
@@ -432,10 +440,33 @@ export default class RepresentationChooser {
 
       // -- AUTO mode --
       let inStarvationMode = false; // == buffer gap too low == panic mode
+
       return observableCombineLatest(clock$, maxAutoBitrate$, deviceEvents$)
         .pipe(
-          map(([ clock, maxAutoBitrate, deviceEvents ]) => {
-            let newBitrateCeil; // bitrate ceil for the chosen Representation
+          withLatestFrom(playbackQualities$),
+          map(([ [clock, maxAutoBitrate, deviceEvents], playbackQualities ]) => {
+            const playableRepresentations = (() => {
+              const qualitativeRepresentations = playbackQualities != null ?
+              representations.filter((representation) => {
+                const repId = representation.id;
+                const playbackQuality = playbackQualities[repId];
+                if (playbackQuality && playbackQuality < ABR_BAN_THRESHOLD) {
+                  banRepresentationFromPlayback(
+                    representation, bannedRepresentations, playbackQuality);
+                }
+                return !bannedRepresentations.some((banned) =>
+                  banned.id === representation.id);
+              }) : representations;
+
+              if (qualitativeRepresentations.length === 0) {
+                log.warn("ABR - not considering stream qualities \
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use string concatenation rather than line continuation. ([typescript:S1516](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1516))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=295&issues=AWaw4ehdLpqqSUlcT5H-&open=AWaw4ehdLpqqSUlcT5H-)

[ISSUE_KEY]: AWaw4ehdLpqqSUlcT5H-